hazelcast/src/main/java/com/hazelcast/datalink/HazelcastDataLink.java

```diff
@@ -0,0 +1,147 @@
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.internal.util.StringUtil;
+import com.hazelcast.map.IMap;
+import com.hazelcast.spi.annotation.Beta;
+
+import javax.annotation.Nonnull;
+import java.util.Collection;
+import java.util.stream.Collectors;
+
+import static com.hazelcast.jet.impl.util.ImdgUtil.asClientConfig;
+import static com.hazelcast.jet.impl.util.ImdgUtil.asClientConfigFromYaml;
+
+/**
+ * Creates a HazelcastInstance that is shared to connect to a remote cluster.
+ * <p>
+ * Set the client XML as property using {@link HazelcastDataLink#CLIENT_XML} as property key or
+ * Set the client YAML as property using {@link HazelcastDataLink#CLIENT_YML} as property key
+ *
+ * @since 5.3
+ */
+@Beta
+public class HazelcastDataLink extends DataLinkBase implements DataLink {
+
+    /**
+     * The constant to be used as property key for XML
+     */
+    public static final String CLIENT_XML = "client_xml";
+
+    /**
+     * The constant to be used as property key for YAML
+     */
+    public static final String CLIENT_YML = "client_yml";
+
+    private final ClientConfig clientConfig;
+
+    /**
+     * The cached client instance
+     */
+    private volatile HazelcastClientProxy proxy;
+
+    public HazelcastDataLink(@Nonnull DataLinkConfig dataLinkConfig) {
+        super(dataLinkConfig);
+        this.clientConfig = buildClientConfig();
+
+        if (dataLinkConfig.isShared()) {
+            HazelcastClientProxy proxy = (HazelcastClientProxy) HazelcastClient.newHazelcastClient(clientConfig);
+            this.proxy = new HazelcastClientProxy(proxy.client) {
+                @Override
+                public void shutdown() {
+                    release();
+                }
+            };
+        }
+    }
+
+    private ClientConfig buildClientConfig() {
+        ClientConfig clientConfig = null;
+        String clientXml = getConfig().getProperty(CLIENT_XML);
+        if (!StringUtil.isNullOrEmpty(clientXml)) {
+            // Read ClientConfig from XML
+            clientConfig = asClientConfig(clientXml);
+        }
+
+        String clientYaml = getConfig().getProperty(CLIENT_YML);
+        if (!StringUtil.isNullOrEmpty(clientYaml)) {
+            // Read ClientConfig from Yaml
+            clientConfig = asClientConfigFromYaml(clientYaml);
+        }
+
+        if (clientConfig == null) {
+            throw new HazelcastException("HazelcastDataLink with name '" + getConfig().getName()
+                    + "' could not be created, provide either client_xml or client_yml property "
+                    + "with the client configuration.");
+        }
+        return clientConfig;
+    }
+
+    @Nonnull
+    @Override
+    public Collection<DataLinkResource> listResources() {
+        HazelcastInstance instance = getClient();
+        try {
+            return instance.getDistributedObjects()
+                           .stream()
+                           .filter(o -> o instanceof IMap)
```

Comment:
It can be like this too. Easier to read. It is upto you
`.filter(IMap.class::isInstance)`